PWGCF/FemtoDream/femtoDreamProducerReducedTask.cxx

```diff
@@ -153,28 +167,66 @@ struct femtoDreamProducerReducedTask {
     }
 
     if (!ConfIsRun3 || (ConfIsRun3 && ConfIsMC)) {
+
       static o2::parameters::GRPObject* grpo = nullptr;
       if (grpo == nullptr) {
         grpo = ccdb->getForTimeStamp<o2::parameters::GRPObject>("GLO/GRP/GRP", timestamp);
         if (grpo == nullptr) {
           LOGF(fatal, "GRP object not found for timestamp %llu", timestamp);
-          return 0;
+          return;
         }
         LOGF(info, "Retrieved GRP for timestamp %llu with magnetic field of %d kG", timestamp, grpo->getNominalL3Field());
       }
       output = 0.1 * (grpo->getNominalL3Field());
     }
-    return output;
+    mMagField = output;
+    mRunNumber = bc.runNumber();
   }
 
-  void process(aod::FemtoFullCollision const& col, aod::BCsWithTimestamps const&, aod::FemtoFullTracks const& tracks) /// \todo with FilteredFullV0s
+  template <typename ParticleType>
+  void fillMCParticle(ParticleType const& particle)
   {
-    // get magnetic field for run
-    auto bc = col.bc_as<aod::BCsWithTimestamps>();
-    if (mRunNumber != bc.runNumber()) {
-      mMagField = getMagneticFieldTesla(bc.timestamp());
-      mRunNumber = bc.runNumber();
+    if (particle.has_mcParticle()) {
+      // get corresponding MC particle and its info
+      auto particleMC = particle.mcParticle();
+      auto pdgCode = particleMC.pdgCode();
+
+      int particleOrigin = 99;
+      auto motherparticleMC = particleMC.template mothers_as<aod::McParticles>().front();
+
+      if (abs(pdgCode) == 2212) { // TODO: create a configurable and do not hardcode!
+        if (particleMC.isPhysicalPrimary()) {
+          particleOrigin = aod::femtodreamparticleMC::ParticleOriginMCTruth::kPrimary;
+        } else if (motherparticleMC.producedByGenerator()) {
+          switch (abs(motherparticleMC.pdgCode())) {
+            case 3122:
+              particleOrigin = aod::femtodreamparticleMC::ParticleOriginMCTruth::kDaughterLambda;
+              break;
+            case 3222:
+              particleOrigin = aod::femtodreamparticleMC::ParticleOriginMCTruth::kDaughterSigmaplus;
+              break;
+            default:
+              particleOrigin = aod::femtodreamparticleMC::ParticleOriginMCTruth::kDaughter;
+          } // switch
+        } else {
+          particleOrigin = aod::femtodreamparticleMC::ParticleOriginMCTruth::kMaterial;
+        }
+      } else {
+        particleOrigin = aod::femtodreamparticleMC::ParticleOriginMCTruth::kFake;
+      }
+      outputPartsMC(particleOrigin, pdgCode, particleMC.pt(), particleMC.eta(), particleMC.phi());
+      // fill with correct values, this is currently placeholder
+      // outputDebugPartsMC(-999);
+    } else {
+      outputPartsMC(-999, -999, -999, -999, -999);
```

Comment:
Should not be safer to set `aod::femtodreamparticleMC::ParticleOriginMCTruth::kFake` instead of `-999`

Reply:
So far the code was only meant to be used for the assumption that the paritcle is a proton and the default values have been set accordingly. That will be updated in the future.